apache/superset
# fix(explore): expired form data for SQL Lab queries

### SUMMARY

#19981 added the ability for Explore to fallback to default Explore form state (based on chart id or datasource id) had cached form data expires. But this feature is not working for Explore from SQL Lab---because of inconsistent URL parameter names. The `/explore` API expects `dataset_id=xxx`, but SQL Lab page added `datasource_id=xxx`.

This PR fixes this issue and added some backward compatibility so that both `datasource_id` and `dataset_id` works.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


#### Before

When visiting Explore from SQL Lab, You will get `missing_datasource` if your explore form data has expired.

<img width="1116" alt="image" src="https://user-images.githubusercontent.com/335541/189445944-36425bc1-1221-4576-a0ac-12717a7461d4.png">

#### After

Explore should load as long as a valid `dataset_id` is provided

<img width="918" alt="image" src="https://user-images.githubusercontent.com/335541/189447544-f2196672-6346-4271-a770-46ecef7eba29.png">



### TESTING INSTRUCTIONS

1. Open Explore from SQL Lab query results
2. Change the `form_data_key` to an invalid string to simulate expired form data key. For example: http://localhost:9000/explore/?form_data_key=invalid&datasource_id=45&datasource_type=query
3. The Explore page should still load.

### ADDITIONAL INFORMATION
